cmd/browser_pools.go

```diff
@@ -0,0 +1,683 @@
+		{"Property", "Value"},
+		{"ID", pool.ID},
+		{"Name", name},
+		{"Size", fmt.Sprintf("%d", pool.BrowserPoolConfig.Size)},
+		{"Available", fmt.Sprintf("%d", pool.AvailableCount)},
+		{"Acquired", fmt.Sprintf("%d", pool.AcquiredCount)},
+		{"Timeout", fmt.Sprintf("%d", pool.BrowserPoolConfig.TimeoutSeconds)},
+		{"Created At", util.FormatLocal(pool.CreatedAt)},
+	}
+	PrintTableNoPad(tableData, true)
+	return nil
+}
+
+type BrowserPoolsUpdateInput struct {
+	IDOrName           string
+	Name               string
+	Size               int64
+	FillRate           int64
+	TimeoutSeconds     int64
+	Stealth            BoolFlag
+	Headless           BoolFlag
+	Kiosk              BoolFlag
+	ProfileID          string
+	ProfileName        string
+	ProfileSaveChanges BoolFlag
+	ProxyID            string
+	Extensions         []string
+	Viewport           string
+	DiscardAllIdle     BoolFlag
+}
+
+func (c BrowserPoolsCmd) Update(ctx context.Context, in BrowserPoolsUpdateInput) error {
+	req := kernel.BrowserPoolUpdateRequestParam{}
+
+	if in.Name != "" {
+		req.Name = kernel.String(in.Name)
+	}
+	if in.Size > 0 {
+		req.Size = in.Size
+	}
+	if in.FillRate > 0 {
+		req.FillRatePerMinute = kernel.Int(in.FillRate)
+	}
+	if in.TimeoutSeconds > 0 {
+		req.TimeoutSeconds = kernel.Int(in.TimeoutSeconds)
+	}
+	if in.Stealth.Set {
+		req.Stealth = kernel.Bool(in.Stealth.Value)
+	}
+	if in.Headless.Set {
+		req.Headless = kernel.Bool(in.Headless.Value)
+	}
+	if in.Kiosk.Set {
+		req.KioskMode = kernel.Bool(in.Kiosk.Value)
+	}
+	if in.DiscardAllIdle.Set {
+		req.DiscardAllIdle = kernel.Bool(in.DiscardAllIdle.Value)
+	}
+
+	// Profile
+	if in.ProfileID != "" && in.ProfileName != "" {
+		pterm.Error.Println("must specify at most one of --profile-id or --profile-name")
+		return nil
+	} else if in.ProfileID != "" || in.ProfileName != "" {
+		req.Profile = kernel.BrowserProfileParam{
+			SaveChanges: kernel.Bool(in.ProfileSaveChanges.Value),
+		}
+		if in.ProfileID != "" {
+			req.Profile.ID = kernel.String(in.ProfileID)
+		} else if in.ProfileName != "" {
+			req.Profile.Name = kernel.String(in.ProfileName)
+		}
+	}
+
+	if in.ProxyID != "" {
+		req.ProxyID = kernel.String(in.ProxyID)
+	}
+
+	// Extensions
+	if len(in.Extensions) > 0 {
+		for _, ext := range in.Extensions {
+			val := strings.TrimSpace(ext)
+			if val == "" {
+				continue
+			}
+			item := kernel.BrowserExtensionParam{}
+			if cuidRegex.MatchString(val) {
+				item.ID = kernel.String(val)
+			} else {
+				item.Name = kernel.String(val)
+			}
+			req.Extensions = append(req.Extensions, item)
+		}
+	}
+
+	// Viewport
+	if in.Viewport != "" {
+		width, height, refreshRate, err := parseViewport(in.Viewport)
+		if err != nil {
+			pterm.Error.Printf("Invalid viewport format: %v\n", err)
+			return nil
+		}
+		req.Viewport = kernel.BrowserViewportParam{
+			Width:  width,
+			Height: height,
+		}
+		if refreshRate > 0 {
+			req.Viewport.RefreshRate = kernel.Int(refreshRate)
+		}
+	}
```

Comment:
maybe there should be a shared helper that Create and Update use for
- Profile validation and construction
- Extensions parsing
- Viewport parsing

Reply:
aye sorry for the slop https://github.com/onkernel/cli/pull/40/commits/1b1935ecea30d69b919364d24933ef5c7d16f2e3